ocaml/xapi/xapi_observer.ml

```diff
@@ -110,14 +110,192 @@ module Observer : ObserverInterface = struct
+    }
+end
+
+module type COMPONENT = sig
+  val component_name : string
+end
+
+module Dom0ObserverConfig (Component : COMPONENT) : ObserverInterface = struct
+  let config_root = "/etc/xensource/observer/"
+
+  let env_vars_of_config (config : ObserverConfig.t) =
+    EnvRecord.(
+      to_shell_string
+        [
+          ("OTEL_SERVICE_NAME", str config.otel_service_name)
+        ; ( "OTEL_RESOURCE_ATTRIBUTES"
+          , (list pair) config.otel_resource_attributes
+          )
+        ; ( "OTEL_EXPORTER_ZIPKIN_ENDPOINTS"
+          , (list str) config.otel_exporter_zipkin_endpoints
+          )
+        ; ( "XS_EXPORTER_BUGTOOL_ENDPOINT"
+          , option config.xs_exporter_bugtool_endpoint
+          )
+        ]
+    )
+
+  let remove_config ~uuid =
+    Xapi_stdext_unix.Unixext.unlink_safe
+      (config_root
+      ^ Component.component_name
+      ^ "/enabled/"
+      ^ uuid
+      ^ ".observer.conf"
+      )
+
+  let update_config ~__context ~observer ~config_root ~uuid =
+    if Db.Observer.get_enabled ~__context ~self:observer then (
+      let observer_config =
+        ObserverConfig.config_of_observer ~__context
+          ~component:Component.component_name ~observer
+      in
+      let dir_name = config_root ^ Component.component_name ^ "/enabled/" in
+      Xapi_stdext_unix.Unixext.mkdir_rec dir_name 0o755 ;
+      let file_name = dir_name ^ uuid ^ ".observer.conf" in
+      Xapi_stdext_unix.Unixext.write_string_to_file file_name
+        (env_vars_of_config observer_config)
+    ) else
+      remove_config ~uuid
+
+  let update_all_configs ~__context ~observer_all =
+    List.iter
+      (fun observer ->
+        let uuid = Db.Observer.get_uuid ~__context ~self:observer in
+        update_config ~__context ~observer ~config_root ~uuid
+      )
+      observer_all
+
+  let create ~__context ~uuid ~name_label:_ ~attributes:_ ~endpoints:_
+      ~enabled:_ =
+    let observer = Db.Observer.get_by_uuid ~__context ~uuid in
+    update_config ~__context ~observer ~config_root ~uuid
+
+  let destroy ~__context ~uuid = remove_config ~uuid
+
+  let set_enabled ~__context ~uuid ~enabled:_ =
+    let observer = Db.Observer.get_by_uuid ~__context ~uuid in
+    update_config ~__context ~observer ~config_root ~uuid
+
+  let set_attributes ~__context ~uuid ~attributes:_ =
+    let observer = Db.Observer.get_by_uuid ~__context ~uuid in
+    update_config ~__context ~observer ~config_root ~uuid
+
+  let set_endpoints ~__context ~uuid ~endpoints:_ =
+    let observer = Db.Observer.get_by_uuid ~__context ~uuid in
+    update_config ~__context ~observer ~config_root ~uuid
+
+  let init ~__context =
+    let observer_all = Db.Observer.get_all ~__context in
+    update_all_configs ~__context ~observer_all
+
+  let set_trace_log_dir ~__context ~dir:_ =
+    let observer_all = Db.Observer.get_all ~__context in
+    update_all_configs ~__context ~observer_all
+
+  let set_export_interval ~__context:_ ~interval:_ = ()
```

Comment:
so where is the export_interval set then?

Reply:
It's done by the Observer (the default Xapi one)